debezium-server-redis/src/main/java/io/debezium/server/redis/RedisStreamChangeConsumer.java

```diff
@@ -226,4 +236,14 @@ else if (isMemoryOk.check()) {
         // Mark the whole batch as finished once the sub batches completed
         committer.markBatchFinished();
     }
-}
\ No newline at end of file
+
+    private long getObjectSize(SimpleEntry<String, Map<String, String>> record) throws IOException {
+        ByteArrayOutputStream boas = new ByteArrayOutputStream();
+        ObjectOutputStream oos = new ObjectOutputStream(boas);
+        oos.writeObject(record);
+        oos.flush();
+        long size = boas.size() + 50;
```

Comment:
What is the magical 50 constant?

Reply:
Created constant. 50 is redis stream memory overhead based on my observation.,